metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_v2.py

```diff
@@ -1608,7 +1622,7 @@ def close(self) -> None:
         StatefulIngestionSourceBase.close(self)
         self.view_definitions.close()
         self.sql_parser_schema_resolver.close()
-        if hasattr(self, "lineage_extractor"):
+        if hasattr(self, "lineage_extractor") and self.lineage_extractor:
```

Comment:
I think we can get rid of the hasattr check here and below

Reply:
yes